src/runtime_src/driver/xclng/xrt/user_gem/perf.cpp

```diff
@@ -178,21 +178,36 @@ namespace xocl {
       return mAccelProfilingNumberSlots;
     if (type == XCL_PERF_MON_STALL)
       return mStallProfilingNumberSlots;
+    if (type == XCL_PERF_MON_STR)
+      return mStreamProfilingNumberSlots;
+
     if (type == XCL_PERF_MON_HOST) {
       uint32_t count = 0;
       for (unsigned int i=0; i < mMemoryProfilingNumberSlots; i++) {
         if (mPerfmonProperties[i] & XSPM_HOST_PROPERTY_MASK) count++;
       }
       return count;
     }
-    if (type == XCL_PERF_MON_STR)
-      return mStreamProfilingNumberSlots;
+    if (type == XCL_PERF_MON_SHELL) {
+      uint32_t count = 0;
+      for (unsigned int i=0; i < mMemoryProfilingNumberSlots; i++) {
+        if (mPerfmonProperties[i] & XSPM_HOST_PROPERTY_MASK) {
```

Comment:
Please use early exit coding style to avoid deep conditional flow--

https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code

Reply:
moved out of range check to beginning